File: poo/python/Interfaces/Conta/Banco.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from ContaCorrente import ContaCorrente
from ContaPoupanca import ContaPoupanca
from CDB import CDB

class Banco():

    # Atributos #
    __dinheiro = 0
    contas = []
    __tributos = 0

    # Registra uma nova conta #
    def Registra(self, conta):
        if not isinstance(conta, ContaPoupanca):
            self.__tributos += conta.calculaTributos()
        self.__dinheiro += conta.getSaldo()

    # Imprime o total do banco #
    def Imprime(self):
        print('Saldo do banco: {:.2f}\nTributos do banco: {:.2f}'
              .format(self.__dinheiro, self.__tributos))


if __name__ == '__main__':

    banco = Banco()

    for i in range(5):
        banco.contas.append(ContaCorrente(i))
        banco.contas.append(ContaPoupanca(i))
        banco.contas.append(CDB(i))

    for i in range(15):
        banco.contas[i].credita(12)
        banco.contas[i].debita(3)
        banco.Registra(banco.contas[i])

    banco.Imprime()

File: poo/java/Collecions/Conta2/Banco.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

import java.text.*;

public class Banco{

    /* Atributos */
    private double dinheiro;
    private Conta[] contas;
    private double tributos;

    /* Construtor */
    public Banco(){
        this.dinheiro = 0.0;
        this.contas = new Conta[15];
        this.tributos = 0.0;
    }

    /* Registra nova conta */
    public void Registra(Conta conta){
        ContaCorrente aux;
        CDB aux2;

        if(conta instanceof ContaCorrente){
            aux = (ContaCorrente) conta;
            this.tributos += aux.calculaTributos();
        }
        else if(conta instanceof CDB){
            aux2 = (CDB) conta;
            this.tributos += aux2.calculaTributos();
        }
        this.dinheiro += conta.getSaldo();
    }

    /* Imprime o total do banco */
    public void Imprime(){
        DecimalFormat casas = new DecimalFormat("#.00");
        System.out.println("Saldo do banco: " + casas.format(this.dinheiro));
        System.out.println("Tributos do banco: " + casas.format(this.tributos));
    }

    public static void main(String args[]){
        Banco banco = new Banco();
        ContaCorrente[] cc = new ContaCorrente[5];
        ContaPoupanca[] cp = new ContaPoupanca[5];
        CDB[] cdb = new CDB[5];
        int i, j;

        for(i=0;i<5;i++){
            cc[i] = new ContaCorrente(i);
            cp[i] = new ContaPoupanca(i);
            cdb[i] = new CDB(i);
        }

        for(i=0,j=0;i<15;i+=3,j++){
            banco.contas[i] = cc[j];
            banco.contas[i+1] = cp[j];
            banco.contas[i+2] = cdb[j];
        }

        for(i=0;i<15;i++){
            banco.contas[i].credita(12);
            banco.contas[i].debita(3);
            banco.Registra(banco.contas[i]);
        }

        banco.Imprime();

    }
}
